### ISSUE TITLE
UNINITIALIZED_VALUE - false positive C

### ISSUE THREAD

**Issue by PatriciaSVMonteiro** [facebook/infer#974]
In the following minimal example, the `var` variable is initialized into a `while(true)`, however, Infer reports an UNINITIALIZED_VALUE. I checked this case with `if (true)`, `for ( ; ; )` and `while(1)` and Infer returns the same error. 
```
int main(){
	int var;
	while(true){
		var = 0;
		if(var==0) break;
	}
	printf("Var:%d\n",var);
	return 0;
}
```
This is the Infer's report:
```
Found 1 issue

while_true.c:12: error: UNINITIALIZED_VALUE
  The value read from var was never initialized.
  10.   		if(var==0) break;
  11.   	}
  12. > 	printf("Var:%d\n",var);
  13.   	return 0;
  14.   }

Summary of the reports

  UNINITIALIZED_VALUE: 1
```

**Comment by jvillard** [comment]
You're right, this checker doesn't track the value of the loop condition so doesn't realise it's always true. Would it be an option to hoist `var = 0` out of the loop ?

**Comment by PatriciaSVMonteiro** [comment]
Thanks for the suggestion, yes in that case Infer would not report an error.
This problem was encountered during the analysis of the SQLite code, Infer reports this false positive in several situations, on the other hand the Clang Static Analyzer realizes that the condition of the loop is always true and therefore does not report any error.
